athena-opensearch/src/main/java/com/amazonaws/athena/connectors/opensearch/OpensearchMetadataHandler.java

```diff
@@ -0,0 +1,426 @@
+                        if (columnName.contains(".")) {
+                            FieldBuilder rootField = schemaBuilder.getNestedField(columnName.split("\\.")[0]);
+                            FieldBuilder newField = OpensearchUtils.createNestedStruct(columnName, rootField, columnType);
+                            schemaBuilder.addNestedField(columnName.split("\\.")[0], newField);
+                        }
+                        else {
+                            schemaBuilder.addField(FieldBuilder.newBuilder(columnName, columnType).build());
+                        }
+                    }
+                }
+                else {
+                    // Default to VARCHAR ArrowType
+                    LOGGER.warn("getSchema: Unable to map type for column[" + columnName +
+                            "] to a supported type, attempted " + columnType + " - defaulting type to VARCHAR.");
+                    schemaBuilder.addField(FieldBuilder.newBuilder(columnName, new ArrowType.Utf8()).build());
+                }
+                found = true;
+            }
+
+            if (!found) {
+                throw new RuntimeException("Could not find table in " + tableName.getSchemaName());
+            }
+
+            // add partition columns
+            partitionSchema.getFields().forEach(schemaBuilder::addField);
+
+            return schemaBuilder.build();
+        }
+    }
+
+    private int decodeContinuationToken(GetSplitsRequest request)
+    {
+        if (request.hasContinuationToken()) {
+            return Integer.valueOf(request.getContinuationToken());
+        }
+
+        //No continuation token present
+        return 0;
+    }
+
+    private String encodeContinuationToken(int partition)
+    {
+        return String.valueOf(partition);
+    }
+
+    @Override
+    public ListTablesResponse doListTables(final BlockAllocator blockAllocator, final ListTablesRequest listTablesRequest)
+            throws Exception
+    {
+        try (Connection connection = jdbcConnectionFactory.getConnection(getCredentialProvider())) {
+            LOGGER.info("{}: List table names for Catalog {}, Table {}", listTablesRequest.getQueryId(), listTablesRequest.getCatalogName(), listTablesRequest.getSchemaName());
+            return new ListTablesResponse(listTablesRequest.getCatalogName(),
+                    listTables(connection, listTablesRequest.getSchemaName()), null);
+        }
+    }
+
+    private List<TableName> listTables(final Connection jdbcConnection, final String databaseName)
+            throws SQLException
+    {
+        try (ResultSet resultSet = getTables(jdbcConnection, databaseName)) {
+            ImmutableList.Builder<TableName> list = ImmutableList.builder();
+            while (resultSet.next()) {
+                // checks if table name is internal table ex: .kibana
+                String tableName = resultSet.getString("TABLE_NAME");
+                if (tableName.startsWith(".")) {
```

Comment:
nit, just combine these two conditionals with &&. it would make sense to keep them separate if we had alternate behavior in the inner conditional, but we do not.